format/FlightSql.proto

```diff
@@ -1549,6 +1658,17 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+/*
+ * Explicitly cancel a running query.
+ *
+ * For distributed queries (ones that use GetFlightInfo->DoGet), this lets a
+ * single client explicitly cancel work across all clients, given server support.
+ */
+message ActionCancelQuery {
+  // The result of the GetFlightInfo RPC that initated the query.
+  FlightInfo info = 1;
```

Comment:
should this be some sort of ID instead?

Reply:
What would it be? Since most of the data inside FlightInfo isn't meant to be introspected, and we haven't specified what the contents of Ticket should be, there's no consistent 'query ID' concept right now. So I chose FlightInfo since presumably that has all information the server needs to identify the query.

Reply:
would this be another place where app_metadata would be useful?  Otherwise, it seems like the only implementation path for the server would be to introspect one of the tickets? (it would seem lfight descriptors might very commonly be non-unique).  This might be fine, I just want to confirm my understanding.

Reply:
yeah, I think introspecting the tickets for now should be fine.